polymetis/polymetis/python/polysim/grpc_sim_client.py

```diff
@@ -183,3 +210,30 @@ def execute_rpc_call(self, request_func: Callable, args=[], log_request_time=Fal
                 self.interval_log = []
 
         return ret
+
+    def init_robot_client(self):
+        self.connection.InitRobotClient(self.metadata.get_proto())
+
+    def set_robot_state(self, robot_state: polymetis_pb2.RobotState):
+        self.env.set_robot_state(robot_state)
+
+    def _sync_blocking(self, tgt_robot: RobotInterface, timesteps: int):
+        step = 0
+        while step < timesteps and not self._kill_state_setter:
+            tgt_state = tgt_robot.get_robot_state()
+            self.set_robot_state(tgt_state)
+            step += 1
+
+    def sync(self, tgt_robot: RobotInterface, timesteps: int = float("inf")):
+        assert (
+            self._state_setter is None
+        ), "The simulation client is already synced to a robot"
+        self._state_setter = Thread(
+            target=self._sync_blocking, args=[tgt_robot, timesteps]
+        )
+        self._state_setter.start()
+
+    def unsync(self):
+        assert self._state_setter is not None, "The mirror simulator is not synced"
+        self._kill_state_setter = True
+        self._state_setter.join()
```

Comment:
The `sync` and `unsync` methods looks reasonable, great job getting the most complicated part in place! However, it is calling `get_robot_state` within a while loop without any proper waiting/spinning, which might overload the server with such frequent requests. Try to run the loop at a fixed, reasonable frequency that can be changed in the [config](https://github.com/facebookresearch/fairo/blob/main/polymetis/polymetis/conf/launch_robot.yaml), by either sleeping for `1. /  mirror_hz` each loop or better yet by using the [Spinner](https://github.com/facebookresearch/fairo/blob/main/polymetis/polymetis/python/polymetis/utils/spinner.py).
I would say 24 hz should be more than enough for most mirroring use cases.

Reply:
Also, `float("inf")` isn't an `int` and might not pass the typing checks. It would be cleaner to simply default to 0 or -1 and have the while run indefinitely if `timesteps <= 0`.